rfd/0016-dynamic-configuration.md

```diff
@@ -0,0 +1,181 @@
+---
+authors: Andrej Tokarčík (andrej@goteleport.com)
+state: draft
+---
+
+# RFD 16 - Dynamic Configuration
+
+## What
+
+Some resources like `services.AuthPreference` or `services.ClusterConfig` are
+live representations of certain parts of the `auth_service` configuration
+section.  These resources can be understood to constitute *dynamic*
+configuration of an auth server, as opposed to *static* configuration defined
+by its `teleport.yaml` file.
+
+The current behaviour is that Teleport creates dynamic configuration
+implicitly by deriving it from static configuration during auth server
+initialization. The recent proposals of exposing the related resources via
+`tctl` would however allow the dynamic configuration to be created/updated
+explicitly by the user, independently of static configuration.
+
+This RFD presents several possible scenarios involving the interaction of
+explicitly-managed dynamic and static configuration, including the respective
+design decisions to be made, with the goal of reaching a consensus on the
+configuration workflows to be supported by Teleport.
+
+## Why
+
+1. Automated/programmatic management of Teleport clusters.
+
+2. Teleport-as-a-service offerings that do not allow direct access
+   to `teleport.yaml` files.
+
+## Scenarios
+
+We use the example of the `authentication` subsection of `auth_service`
+in `teleport.yaml` (as "the" static configuration) and the corresponding
+resource `services.AuthPreference` (as "the" dynamic configuration).
+The latter is assumed to be manipulable via `tctl` using the identifier `cap`.
+
+### Scenario 1: static configuration specified
+
+1. The `authentication` section is specified in `teleport.yaml`.
+   The backend can have any `AuthPreference` data stored from the last
+   auth server run.
+
+2. The auth server initialization procedure MUST commit data derived from the
+   static configuration to the backend, overwriting any potential resource data
+   already stored there.
+
+3. `tctl get cap` returns the resource data in line with `teleport.yaml`.
+
+Therefore, static configuration is always given precedence.  This should also
+ensure backward compatibility with the already established workflows.
+
+### Scenario 2: static never specified & dynamic not updated by user
```

Comment:
I really like the decision tree layout in this RFD @andrejtokarcik
Really helps to  describe all possible permutations and edge cases